test-project/Assets/EditmodeTests/Utility/InterestQueryTests.cs

```diff
@@ -0,0 +1,46 @@
+using Improbable.Gdk.QueryBasedInterest;
+using NUnit.Framework;
+
+namespace Improbable.Gdk.EditmodeTests.Utility
+{
+    [TestFixture]
+    public class InterestQueryTests
+    {
+        private static InterestQuery BasicQuery => InterestQuery.Query(Constraint.Component(Position.ComponentId));
+
+        [Test]
+        public void Query_sets_FullSnapshotResult_to_true()
+        {
+            ComponentInterest.Query query = BasicQuery;
+            Assert.True(query.FullSnapshotResult);
+        }
+
+        [Test]
+        public void Query_sets_ResultComponentId_to_empty_list()
+        {
+            ComponentInterest.Query query = BasicQuery;
+            Assert.AreEqual(0, query.ResultComponentId.Count);
+        }
+
+        [Test]
+        public void Frequency_null_if_MaxFrequencyHz_never_called()
+        {
+            ComponentInterest.Query query = BasicQuery;
+            Assert.IsNull(query.Frequency);
+        }
+
+        [Test]
+        public void Filter_with_no_components_does_nothing()
+        {
+            var query = BasicQuery.FilterResults();
+            Assert.AreEqual(0, query.ResultComponentId.Count);
+        }
+
+        [Test]
+        public void Filter_with_components_implies_FullSnapshotResult_is_null()
+        {
+            var query = BasicQuery.FilterResults(Position.ComponentId);
+            Assert.IsNull(query.FullSnapshotResult);
```

Comment:
Why is this `null` instead of false ?

Reply:
both null and false act as the 0 value - and since only one of the fields out of `result_component_ids` and `full_snapshot_result` needs to be set, I thought it would be better to explicitly set this to null